rpc/mockSearcher/mock_searcher.go

```diff
@@ -0,0 +1,37 @@
+package mockSearcher
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/zoekt"
+	"github.com/google/zoekt/query"
+)
+
+type MockSearcher struct {
+	WantSearch   query.Q
+	SearchResult *zoekt.SearchResult
+
+	WantList query.Q
+	RepoList *zoekt.RepoList
+}
+
+func (s *MockSearcher) Search(ctx context.Context, q query.Q, opts *zoekt.SearchOptions) (*zoekt.SearchResult, error) {
+	if q.String() != s.WantSearch.String() {
+		return nil, fmt.Errorf("got query %s != %s", q.String(), s.WantSearch.String())
+	}
+	return s.SearchResult, nil
+}
+
+func (s *MockSearcher) List(ctx context.Context, q query.Q) (*zoekt.RepoList, error) {
+	if q.String() != s.WantList.String() {
+		return nil, fmt.Errorf("got query %s != %s", q.String(), s.WantList.String())
+	}
+	return s.RepoList, nil
+}
+
+func (*MockSearcher) Close() {}
+
+func (*MockSearcher) String() string {
+	return "MockSearcher"
+}
```

Comment:
this is quite a small package. Why did you create and export it?

Reply:
I didn't want to copy mockSearcher to the stream package. It seems useful enough to be reused elsewhere. We could move it to an internal package?